central/processlisteningonport/store/postgres/full_store.go

```diff
@@ -34,9 +34,9 @@ type fullStoreImpl struct {
 const getByDeploymentStmt = "SELECT plop.id, plop.serialized, " +
 	"proc.serialized as proc_serialized " +
 	"FROM process_listening_on_ports plop " +
-	"JOIN process_indicators proc " +
+	"LEFT OUTER JOIN process_indicators proc " +
 	"ON plop.processindicatorid = proc.id " +
-	"WHERE proc.deploymentid = $1 AND plop.closed = false"
+	"WHERE plop.deploymentid = $1 AND plop.closed = false"
```

Comment:
We need to use the deploymentid from plop as proc may be NULL as a result of the left outer join.

Reply:
Try to optimize query.

Reply:
Again, first show there are any concerns exist about this query.

Reply:
agree with @erthalion the query isn't horribly complicated though it could involve a lot of data.  But unless you are seeing problems I'm not sure it is worth spending much time on.